contracts/v2/marketplace/TokenMarketplaceV2.sol

```diff
@@ -0,0 +1,509 @@
+pragma solidity ^0.4.24;
+
+import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
+import "openzeppelin-solidity/contracts/access/Whitelist.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+interface IKODAV2Methods {
+  function ownerOf(uint256 _tokenId) external view returns (address _owner);
+
+  function exists(uint256 _tokenId) external view returns (bool _exists);
+
+  function editionOfTokenId(uint256 _tokenId) external view returns (uint256 tokenId);
+
+  function artistCommission(uint256 _tokenId) external view returns (address _artistAccount, uint256 _artistCommission);
+
+  function editionOptionalCommission(uint256 _tokenId) external view returns (uint256 _rate, address _recipient);
+
+  function safeTransferFrom(address _from, address _to, uint256 _tokenId) external;
+}
+
+// Based on ITokenMarketplace.sol
+contract TokenMarketplaceV2 is Whitelist, Pausable {
+  using SafeMath for uint256;
+
+  event UpdatePlatformPercentageFee(uint256 _oldPercentage, uint256 _newPercentage);
+  event UpdateRoyaltyPercentageFee(uint256 _oldPercentage, uint256 _newPercentage);
+  event UpdateMinBidAmount(uint256 minBidAmount);
+
+  event TokenListed(
```

Comment:
`TokenListed`, `TokenDeListed` etc. is also in the `ITokenListingMarketplace` interface. Why not implement the interface?

Reply:
if implementing the interface, other events like bid placed could be added

Reply:
Fixed